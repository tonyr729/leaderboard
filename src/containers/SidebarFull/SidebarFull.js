import React from 'react';
import titleLogo from '../../assets/Leaderboard-Type-Logo-FLAT.svg';
import '../../fontello/css/fontello.css';
import './SidebarFull.css';

const SidebarFull = () => {
  return (
    <div className="sidebar-full">
      <img className="sidebar__title"src={titleLogo} alt="title logo"/>
      <section className="sidebar__section-main">
        <h3>Main</h3>
        <button><i className="icon-chart-bar"></i> Events</button>
        <button><i className="icon-user"></i> Riders</button>
        <button><i className="icon-youtube-play"></i> Media</button>
      </section>
      <section className="sidebar__section-admin">
        <h3>Admin</h3>
        <button disabled><i className="icon-plus"></i> Update Leaderboard</button>
        <button disabled><i className="icon-chart-bar"></i> Create Event</button>
        <button disabled><i className="icon-user"></i> Add Riders</button>
        <button disabled><i className="icon-youtube-play"></i> Add Media</button>
      </section>
    </div>
  )
}

export default SidebarFull;