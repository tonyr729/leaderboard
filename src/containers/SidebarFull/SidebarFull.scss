.sidebar-container {
  display: flex;
  .hidden {
    display: none;
  }
  .sidebar-full {
    display: -ms-flexbox;
    display: flex;
    flex-direction: column;
    align-items: center;
    .sidebar__title{
      width: 80%;
      margin-top: 1.5rem;
    }
    h3 {
      font-weight: 100;
      color: #fff;
    }
    
    .sidebar__btn-admin {
      height: 50px;
      width: 20vw;
      transition: width .5s ease-in-out;
      position: fixed;
      bottom: 0;
      font-size: 2rem;
      left: 0;
      background-color: #5275a7;
      padding: 0;
      color: white;
      border: 0;
    }

    section {
      display: flex;
      flex-direction: column;
      width: 80%;
      margin-top: 50px;
      button {
        margin: .7rem 0;
        width: 100%;
        padding: 1rem;
        text-align: start;
        font-size: 1rem;
        font-weight: 100;
        border: none;
        border-radius: .2rem;
        background: none;
        color: #bbb;
        &:hover {
          background: #6a98da;
          color: #fff;
        }
      }
    }

    .sidebar__section-admin {
      h3 {
        color: #bbbbbb3b;
      }
      button {
        &:disabled {
          color: #bbbbbb3b;
          &:hover {
            background: none;
          }
        }
      }
    }
  }
  .sidebard-collapse {
    padding-top: 1.5rem;
    font-size: 2rem;
    color: #6a98da;
    background: #f7f6f6;
  }
}
  