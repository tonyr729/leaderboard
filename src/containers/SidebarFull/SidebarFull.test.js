import React from "react";
import { shallow } from "enzyme";
import SidebarFull from "./SidebarFull";

describe('SidebarFull', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SidebarFull />);
    expect(wrapper).toMatchSnapshot();
  });
});