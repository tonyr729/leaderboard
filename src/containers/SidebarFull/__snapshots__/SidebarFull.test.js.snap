// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarFull should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarFull />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="title logo"
          className="sidebar__title"
          src="Leaderboard-Type-Logo-FLAT.svg"
/>,
        <section
          className="sidebar__section-main"
>
          <h3>
                    Main
          </h3>
          <button>
                    <i
                              className="icon-chart-bar"
                    />
                     Events
          </button>
          <button>
                    <i
                              className="icon-user"
                    />
                     Riders
          </button>
          <button>
                    <i
                              className="icon-youtube-play"
                    />
                     Media
          </button>
</section>,
        <section
          className="sidebar__section-admin"
>
          <h3>
                    Admin
          </h3>
          <button
                    disabled={true}
          >
                    <i
                              className="icon-plus"
                    />
                     Update Leaderboard
          </button>
          <button
                    disabled={true}
          >
                    <i
                              className="icon-chart-bar"
                    />
                     Create Event
          </button>
          <button
                    disabled={true}
          >
                    <i
                              className="icon-user"
                    />
                     Add Riders
          </button>
          <button
                    disabled={true}
          >
                    <i
                              className="icon-youtube-play"
                    />
                     Add Media
          </button>
</section>,
      ],
      "className": "sidebar-full",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "title logo",
          "className": "sidebar__title",
          "src": "Leaderboard-Type-Logo-FLAT.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Main
</h3>,
            <button>
              <i
                            className="icon-chart-bar"
              />
               Events
</button>,
            <button>
              <i
                            className="icon-user"
              />
               Riders
</button>,
            <button>
              <i
                            className="icon-youtube-play"
              />
               Media
</button>,
          ],
          "className": "sidebar__section-main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Main",
            },
            "ref": null,
            "rendered": "Main",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="icon-chart-bar"
/>,
                " Events",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-chart-bar",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Events",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="icon-user"
/>,
                " Riders",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-user",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Riders",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="icon-youtube-play"
/>,
                " Media",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-youtube-play",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Media",
            ],
            "type": "button",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Admin
</h3>,
            <button
              disabled={true}
>
              <i
                            className="icon-plus"
              />
               Update Leaderboard
</button>,
            <button
              disabled={true}
>
              <i
                            className="icon-chart-bar"
              />
               Create Event
</button>,
            <button
              disabled={true}
>
              <i
                            className="icon-user"
              />
               Add Riders
</button>,
            <button
              disabled={true}
>
              <i
                            className="icon-youtube-play"
              />
               Add Media
</button>,
          ],
          "className": "sidebar__section-admin",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Admin",
            },
            "ref": null,
            "rendered": "Admin",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="icon-plus"
/>,
                " Update Leaderboard",
              ],
              "disabled": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-plus",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Update Leaderboard",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="icon-chart-bar"
/>,
                " Create Event",
              ],
              "disabled": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-chart-bar",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Create Event",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="icon-user"
/>,
                " Add Riders",
              ],
              "disabled": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-user",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Add Riders",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="icon-youtube-play"
/>,
                " Add Media",
              ],
              "disabled": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon-youtube-play",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Add Media",
            ],
            "type": "button",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="title logo"
            className="sidebar__title"
            src="Leaderboard-Type-Logo-FLAT.svg"
/>,
          <section
            className="sidebar__section-main"
>
            <h3>
                        Main
            </h3>
            <button>
                        <i
                                    className="icon-chart-bar"
                        />
                         Events
            </button>
            <button>
                        <i
                                    className="icon-user"
                        />
                         Riders
            </button>
            <button>
                        <i
                                    className="icon-youtube-play"
                        />
                         Media
            </button>
</section>,
          <section
            className="sidebar__section-admin"
>
            <h3>
                        Admin
            </h3>
            <button
                        disabled={true}
            >
                        <i
                                    className="icon-plus"
                        />
                         Update Leaderboard
            </button>
            <button
                        disabled={true}
            >
                        <i
                                    className="icon-chart-bar"
                        />
                         Create Event
            </button>
            <button
                        disabled={true}
            >
                        <i
                                    className="icon-user"
                        />
                         Add Riders
            </button>
            <button
                        disabled={true}
            >
                        <i
                                    className="icon-youtube-play"
                        />
                         Add Media
            </button>
</section>,
        ],
        "className": "sidebar-full",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "title logo",
            "className": "sidebar__title",
            "src": "Leaderboard-Type-Logo-FLAT.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Main
</h3>,
              <button>
                <i
                                className="icon-chart-bar"
                />
                 Events
</button>,
              <button>
                <i
                                className="icon-user"
                />
                 Riders
</button>,
              <button>
                <i
                                className="icon-youtube-play"
                />
                 Media
</button>,
            ],
            "className": "sidebar__section-main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Main",
              },
              "ref": null,
              "rendered": "Main",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="icon-chart-bar"
/>,
                  " Events",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-chart-bar",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Events",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="icon-user"
/>,
                  " Riders",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-user",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Riders",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="icon-youtube-play"
/>,
                  " Media",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-youtube-play",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Media",
              ],
              "type": "button",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Admin
</h3>,
              <button
                disabled={true}
>
                <i
                                className="icon-plus"
                />
                 Update Leaderboard
</button>,
              <button
                disabled={true}
>
                <i
                                className="icon-chart-bar"
                />
                 Create Event
</button>,
              <button
                disabled={true}
>
                <i
                                className="icon-user"
                />
                 Add Riders
</button>,
              <button
                disabled={true}
>
                <i
                                className="icon-youtube-play"
                />
                 Add Media
</button>,
            ],
            "className": "sidebar__section-admin",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Admin",
              },
              "ref": null,
              "rendered": "Admin",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="icon-plus"
/>,
                  " Update Leaderboard",
                ],
                "disabled": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-plus",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Update Leaderboard",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="icon-chart-bar"
/>,
                  " Create Event",
                ],
                "disabled": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-chart-bar",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Create Event",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="icon-user"
/>,
                  " Add Riders",
                ],
                "disabled": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-user",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Add Riders",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="icon-youtube-play"
/>,
                  " Add Media",
                ],
                "disabled": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon-youtube-play",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Add Media",
              ],
              "type": "button",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
